doc/costFunctions.yaml

```diff
@@ -0,0 +1,405 @@
+    The `computeCavitation` flag must be set to `True` because this introduces additional computations involving global communications across processors;
+    If `False`, the returned value is zero.
+    You probably will specify the surface family with `cgnsutilities`.
+    Units: None 
+
+mdot:
+  desc: >
+    Mass flow rate in the freestream direction.
+    This quantity is important for computing momentum forces.
+    Units: kg - s:math:`^{-1}`
+
+mavgptot:
+  desc: >
+    Mass flow rate averaged total pressure 
+    # TODO: GN- I have no idea if this is right based on the source code --> surfaceIntegrations.F90 --> search for 'mass_Ptot'
+    Units: Pascal
+
+aavgptot:
+  desc: >
+    Cell area averaged total pressure 
+    Units: Pascal
+
+aavgps:
+  desc: >
+    Cell area averaged static pressure
+    Units: Pascal
+
+mavgttot:
+  desc: >
+    Mass flow rate averaged total temperature 
+    # TODO: GN- I have no idea if this is right based on the source code --> surfaceIntegrations.F90 --> search for 'mass_Ttot'
+    Units: Kelvin
+
+mavgps:
+  desc: >
+    Mass flow rate averaged static pressure
+    Units: Pascal
+
+mavgmn:
+  desc: >
+    Mass flow rate averaged Mach number
+    Units: None
+
+area:
+  desc: >
+    The area of the integrated surface. 
+    This can be all wall surface or the interested surface defined in the runscript.
+
+axismoment:
+  desc: >
+    Moment of the geometry in the given axis.
+
+flowpower:
+  desc: >
+    Added power by actuator zone to the flow (f.V).
+
+forcexpressure:
+  desc: >
+    Pressure component of force in x direction
+
+forceypressure:
+  desc: >
+    Pressure component of force in y direction
+
+forcezpressure:
+  desc: >
+    Pressure component of force in z direction
+
+forcexviscous:
+  desc: >
+    Viscous component of force in x direction
+
+forceyviscous:
+  desc: >
+    Viscous component of force in y direction
+
+forcezviscous:
+  desc: >
+    Viscous component of force in z direction
+
+forcexmomentum:
+  desc: >
+    Momentum component of force in x direction
+
+forceymomentum:
+  desc: >
+    Momentum component of force in y direction
+
+forcezmomentum:
+  desc: >
+    Momentum component of force in z direction
+
+dragpressure:
+  desc: >
+    Pressure drag
+
+dragviscous:
+  desc: >
+    Viscous drag
+
+dragmomentum:
+  desc: >
+    Momentum deficit drag?
+
+liftpressure:
+  desc: >
+    Pressure component of lift
+
+liftviscous:
+  desc: >
+    Viscous lift
+
+liftmomentum:
+  desc: >
+    Momentum lift (due to changing momentum of flow?)
+
+mavgvx:
+  desc: >
+    Averaged mass flow in the x direction.
+
+mavgvy:
+  desc: >
+    Averaged mass flow in the y direction.
+
+mavgvz:
+  desc: >
+    Averaged mass flow in the z direction.
+
+cperror2:
+  desc: >
+    The square of the different betwween computed cp and target cp*.
```

Comment:
_difference_

We should probably at a minimum link to the function `setTargetCp`. This I believe is for performing inverse design where we want to match a given cp distribution. This would be the L2 norm of the error which we can directly feed into the optimizer as the objective.